src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -605,20 +603,28 @@ bool SotaUptaneClient::getNewTargets(std::vector<Uptane::Target> *new_targets) {
 bool SotaUptaneClient::downloadImages(const std::vector<Uptane::Target> &targets) {
   // Uptane step 4 - download all the images and verify them against the metadata (for OSTree - pull without
   // deploying)
+  std::vector<Uptane::Target> downloaded_targets;
   for (auto it = targets.cbegin(); it != targets.cend(); ++it) {
     // TODO: delegations
     auto images_target = images_repo.getTarget(*it);
     if (images_target == nullptr) {
       LOG_ERROR << "No matching target in images targets metadata for " << *it;
       continue;
+    } else {
+      downloaded_targets.push_back(*it);
```

Comment:
No need for `else`, there is a `continue` in the other condition. We could keep it but then, no reason for line 614 and 618 to sit in different code blocks

Reply:
Fair enough. Fixed.